sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -475,8 +475,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
                 // already pull out those nondeterministic expressions and evaluate them in
                 // a Project node.
                 expr.failAnalysis(
-                  errorClass = "_LEGACY_ERROR_TEMP_2426",
-                  messageParameters = Map("sqlExpr" -> expr.sql))
+                  errorClass = "GROUP_EXPRESSION_IS_NON_DETERMINISTIC",
```

Comment:
Is it user facing error? The comment above says it is an internal one. If not, please, add a test.